README.md

```diff
@@ -470,6 +470,118 @@ class App {
 }
 ```
 
+### Side-Effect Management Using Epics
+
+Ng2-Redux also works well with the `Epic` feature of
+[redux-observable](https://github.com/redux-observable). For
+example, a common use case for a side-effect is making an API call; while
+we can use asynchronous actions for this, epics provide a much cleaner
+approach.
+
+Consider the following example of a user login implementation. First, we
+create some trivial actions:
+
+**session.actions.ts:**
+
+```typescript
+import { Injectable } from '@angular/core';
+import { NgRedux } from 'ng2-redux';
+import { IAppState } from '../reducers';
+
+@Injectable()
+export class SessionActions {
+  static LOGIN_USER = 'LOGIN_USER';
+  static LOGIN_USER_SUCCESS = 'LOGIN_USER_SUCCESS';
+  static LOGIN_USER_ERROR = 'LOGIN_USER_ERROR';
+  static LOGOUT_USER = 'LOGOUT_USER';
+
+  constructor(private ngRedux: NgRedux<IAppState>) {}
+
+  loginUser(credentials) {
+    this.ngRedux.dispatch({
+      type: SessionActions.LOGIN_USER,
+      payload: credentials,
+    });
+  };
+
+  logoutUser() {
+    this.ngRedux.dispatch({ type: SessionActions.LOGOUT_USER });
+  };
+} 
+```
+
+Next, we create an `@Injectable SessionEpic` service:
+
+**session.epics.ts:**
+
+```typescript
+import { Injectable } from '@angular/core';
+import { Http } from '@angular/http';
+import { ActionsObservable } from 'redux-observable';
+import { SessionActions } from '../actions/session.actions';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/observable/of';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/catch';
+
+const BASE_URL = '/api';
+
+@Injectable()
+export class SessionEpics {
+  constructor(private http: Http) {}
+
+  login = (action$: ActionsObservable) => {
+    return action$.ofType(SessionActions.LOGIN_USER)
+      .flatMap(({payload}) => {
```

Comment:
I am having problems using `flatMap` is this still part of `rxjs` API? If not, then seems like this example is a bit outdated.  Otherwise I am missing something else, that is not in this example